text/static/text/js/toolbar.js

```diff
@@ -30,24 +32,22 @@
         } else {
             toolbar.addClass("djtext_toggle");
             body.css('overflow', 'hidden');
+            alert_message.removeClass("view");
         }
         toolbar_active = !toolbar_active;
-    }
+    }   
+
+    form.on('input propertychange paste', function() {
+        changes++;
+        submit.show();
+    });
 
     submit.click(function() {
         form.submit();
         toggle_toolbar();
         submit.hide();
-    });
-
-    form.on('input propertychange paste', function() {
-        changes++;
-        if (changes > 1) {
-            message.text(changes + " changes made during this session");
-        } else {
-            message.text("1 change made during this session");
-        }
-        submit.show();
+        sessions[sessions.length] = changes; 
+        alert_message.addClass("view");
```

Comment:
`alert_message.addClass`, `toggle_toolbar` and `submit.hide` should be called inside the `success` callback in `save_form` so that feedback is given on the actual save, not when you click the button. There should probably also be some feedback when you click the button, like changing the text of it to "Saving ..." or a loader.